src/components/organisms/NavBarSchedule/NavBarSchedule.tsx

```diff
@@ -97,21 +112,93 @@ export const NavBarSchedule: React.FC<NavBarScheduleProps> = ({
     toggle: togglePersonalisedSchedule,
   } = useShowHide(false);
 
+  const liveAndFutureEvents = useMemo(
+    () => relatedVenueEvents.filter(isEventLiveOrFuture),
+    [relatedVenueEvents]
+  );
+  const hasSavedEvents = liveAndFutureEvents
+    .map(
+      prepareForSchedule({
+        usersEvents: userEventIds,
+      })
+    )
+    .filter((event) => event.isSaved).length;
+
+  const isShowPersonalDownloadBtn = hasSavedEvents && showPersonalisedSchedule;
+  const liveEventsMinimalStartValue = Math.min(
+    ...liveAndFutureEvents.map((event) => event.start_utc_seconds)
+  );
+
+  const firstLiveEvent = liveAndFutureEvents.find(
+    (event) => event.start_utc_seconds === liveEventsMinimalStartValue
+  );
+
+  const minDate = useMemo(
+    () =>
+      firstLiveEvent
+        ? isEventLiveOrFuture(firstLiveEvent)
+          ? new Date(liveEventsMinimalStartValue).getTime()
+          : millisecondsToSeconds(todaysDateTime)
+        : minRangeValue,
+    [firstLiveEvent, liveEventsMinimalStartValue]
+  );
+
+  const firstRangeDayTimeValue = isDateLessOrEqualsToday({
+    dateValue: secondsToMilliseconds(minDate),
+    targetDateValue: millisecondsToSeconds(startOfToday().getTime()),
+  });
+
+  const firstRangeDateInSeconds = getUnixTime(
+    max([
+      new Date(secondsToMilliseconds(minDate)).getTime(),
+      todaysDate.getTime(),
+    ])
+  );
+
+  const maxDate = useMemo(
+    () =>
+      Math.max(
+        ...liveAndFutureEvents.map(
+          (event) =>
+            event.start_utc_seconds + minutesToSeconds(event.duration_minutes)
+        ),
+        // + 1 is needed to form a `daysInBetween` timeline and mitigate possible range error
+        firstRangeDateInSeconds + 1
+      ),
+    [liveAndFutureEvents, firstRangeDateInSeconds]
+  );
+
+  const daysInBetween = differenceInDays(
+    fromUnixTime(maxDate),
+    fromUnixTime(firstRangeDateInSeconds)
+  );
+  const dayDifference = getEventDayRange(daysInBetween);
+
   const weekdays = useMemo(() => {
     const formatDayLabel = (day: Date | number) => {
       if (isScheduleTimeshifted) {
-        return format(day, "E, LLL d");
+        return format(day, "do");
       } else {
         return formatDateRelativeToNow(day, {
-          formatOtherDate: (dateOrTimestamp) => format(dateOrTimestamp, "E"),
+          formatOtherDate: (dateOrTimestamp) => format(dateOrTimestamp, "do"),
+          formatTomorrow: (dateOrTimestamp) => format(dateOrTimestamp, "do"),
         });
       }
     };
 
-    return range(SCHEDULE_SHOW_DAYS_AHEAD).map((dayIndex) => {
-      const day = addDays(firstDayOfSchedule, dayIndex);
+    return range(dayDifference).map((dayIndex) => {
+      const firstScheduleDate = firstRangeDayTimeValue
```

Comment:
is `firstRangeDayTimeValue` going to be undefined or null in any case? or it's just zero-check? if so, could it be more explicit? `firstDayTimeValue === 0` ?

Reply:
not going to be undefined, it returns boolean in all cases, but I've renamed it in 60ff2eb, so that the naming is more explicit